lib/vendor.rb

```diff
@@ -1,3 +1,27 @@
+require 'pry'
 class Vendor
-  
+    attr_reader :name,
+                :inventory
+
+    def initialize(name)
+        @name      = name
+        @inventory = {}
+    end
+
+    def check_stock(item)
+        @inventory[item] || 0
+    end
+
+    def stock(item,quantity = 0)
+        @inventory[item] = check_stock(item) + quantity
+    end
+
+    def potential_revenue
+        total_revenue = 0
+        @inventory.each do |item, quantity|
+            price = item.price.delete('$').to_f 
```

Comment:
You cleaned the string and converted the data type here - nice work! This should really live in the item class though.

Reply:
This is a clean enum method otherwise :)

Reply:
Got it! thanks for the call out Nick!